src/Ng/Feed2Catalog.cs

```diff
@@ -240,6 +354,49 @@ private static DateTime ForceUTC(DateTime date)
             return lastDate;
         }
 
+        static async Task<DateTime> Deletes2Catalog(SortedList<DateTime, IList<Tuple<string, string>>> packages, Storage storage, DateTime lastCreated, DateTime lastEdited, DateTime lastDeleted, bool? createdPackages, CancellationToken cancellationToken)
+        {
+            AppendOnlyCatalogWriter writer = new AppendOnlyCatalogWriter(storage, 550);
+
+            DateTime lastDate = createdPackages.HasValue ? (createdPackages.Value ? lastCreated : lastEdited) : DateTime.MinValue;
```

Comment:
break it down to if/else statements. nested ternaries are just impossible to read and maintain